syntax = "proto3";

package master;

service Master {
    rpc Heartbeat(HeartbeatRequest) returns (EmptyResponse) {}
    rpc ReportTaskCompletion(ReportTaskCompletionRequest) returns (EmptyResponse) {}
}

message HeartbeatRequest {
    string ip = 1;
    int32 port = 2;
}

message ReportTaskCompletionRequest {
    string task_id = 1;
    string ip = 2;
    int32 port = 3;
    repeated string output_file_paths = 4;
    string role = 5;
    string work = 6;
}

message EmptyResponse {}